zfit/core/constraint.py

```diff
@@ -128,10 +129,10 @@ def sample(self, n: ztyping.nSamplingTypeIn = 1) -> Dict:
         return {p: sample[:, i] for i, p in enumerate(self.get_params())}
 
     def _sample(self, n):
-        return self.distribution.sample(n)
+        return zfit.run(self.distribution.sample(n))
```

Comment:
We need to cache that in the future at least. Currently, the caching is not optimal and should be easier. Can you just mark it with a `# TODO cache: add proper caching`?

The drawback is not hard, just that _if used inside a loop_, it will become slow. But that's anyway not how it should be used.